spec/services/schools/assign_mentor_spec.rb

```diff
@@ -1,34 +1,76 @@
 
-    it 'adds a new mentorship for the ect with the new mentor started the date given' do
-      expect(ECTAtSchoolPeriods::Mentorship.new(mentee).current_mentor).to eq(current_mentor)
-      expect { service.assign! }.to change(MentorshipPeriod, :count).from(1).to(2)
-      expect(ECTAtSchoolPeriods::Mentorship.new(mentee).current_mentor).to eq(new_mentor)
-      expect(ECTAtSchoolPeriods::Mentorship.new(mentee).current_mentorship_period.started_on).to eq(started_on)
+      it 'ends current mentorship of the ect' do
+        expect { service.assign! }.to change { current_mentorship.reload.finished_on }.from(nil).to(mentorship_started_on)
+      end
+
+      it 'adds a new mentorship for the ect with the new mentor started the date given' do
+        expect(ECTAtSchoolPeriods::Mentorship.new(mentee).current_mentor).to eq(current_mentor)
+        expect { service.assign! }.to change(MentorshipPeriod, :count).from(1).to(2)
+        expect(ECTAtSchoolPeriods::Mentorship.new(mentee.reload).current_mentor).to eq(new_mentor)
+        expect(ECTAtSchoolPeriods::Mentorship.new(mentee).current_mentorship_period.started_on).to eq(mentorship_started_on)
+      end
+
+      context "when no start date is provided" do
+        subject(:service) { described_class.new(author:, ect: mentee, mentor: new_mentor) }
+
+        it "the current date is assigned" do
+          service.assign!
+
+          expect(ECTAtSchoolPeriods::Mentorship.new(mentee.reload).current_mentorship_period.started_on).to eq(Date.current)
+        end
+      end
     end
 
-    context "when no start date is provided" do
-      subject(:service) { described_class.new(author:, ect: mentee, mentor: new_mentor) }
+    describe 'future dates' do
+      context 'when the mentee (ECT) start date is in the future' do
+        let(:mentorship_started_on) { nil }
+        let(:mentee_started_on) { 3.weeks.from_now.to_date }
 
-      it "the current date is assigned" do
-        service.assign!
+        it 'sets the start date to the mentee start date' do
+          service.assign!
+
+          expect(service.mentorship_period.started_on).to eq(mentee_started_on)
+        end
+      end
+
+      context 'when the mentor start dates is in the future' do
+        let(:mentorship_started_on) { nil }
+        let(:mentor_started_on) { 1.month.from_now.to_date }
+
+        it 'sets the start date to the mentee start date' do
+          service.assign!
+
+          expect(service.mentorship_period.started_on).to eq(mentor_started_on)
+        end
+      end
+
+      context 'when the mentee and mentor start dates are in the future but the provided start date is further in the future' do
+        let(:mentorship_started_on) { 2.months.from_now.to_date }
+        let(:mentor_started_on) { 1.month.from_now.to_date }
+        let(:mentee_started_on) { 3.weeks.from_now.to_date }
+
+        it 'sets the start date to the mentee start date' do
```

Comment:
And is this supposed to be this?

```suggestion
        it 'sets the start date to the mentorship start date' do
```